packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -222,13 +222,18 @@ export default class CashierStore extends BaseStore {
             // show p2p if:
             // 1. we have not already checked this before, and
             // 2. client is not virtual
+            // 3. client is not EU user.
             if (!this.is_p2p_visible && !this.root_store.client.is_virtual) {
                 const advertiser_info = await WS.authorized.p2pAdvertiserInfo();
                 const advertiser_error = getPropertyValue(advertiser_info, ['error', 'code']);
                 if (advertiser_error === 'PermissionDenied') this.setIsP2pVisible(true);
 
                 this.is_p2p_advertiser = !advertiser_error;
                 this.setIsP2pVisible(true);
+                when(
+                    () => this.root_store.client.is_eu,
+                    () => this.setIsP2pVisible(false)
+                );
```

Comment:
If I'm not mistaken `onMountCommon` should be executed on each account switch (see `accountSwitcherListener` and look for `onRemount`) so we don't need to use a `when` here, instead we could just expand on the if statement above and save a `p2p_advertiser_info` request in the process.

- Note: The check for `PermissionDenied` looks redundant though, so you may be able to remove it completely.

```JS
if (this.root_store.client.is_eu) {
    this.setIsP2pVisible(false);
} else {
    const advertiser_info = await WS.authorized.p2pAdvertiserInfo();
    const advertiser_error = getPropertyValue(advertiser_info, ['error', 'code']);
    
    if (advertiser_error === 'PermissionDenied') {
        this.setIsP2pVisible(true);
    }

    this.is_p2p_advertiser = !advertiser_error;
    this.setIsP2pVisible(true);
}
```

Reply:
@aaron-binary I used this approach initially but the issue is at that moment the if statement is calculated the `is_eu` flag is not filled. the `is_eu` is calculated based on `landing_companies` that is not ready at that moment.
since this block is running once on the mount and is not reactive, we always get `false` for `is_eu`.
so I had to use `when` to watch for its change in the future. but like you, I don't like this approach too!